### PR TITLE
Use user information instead of auth token on user change

### PR DESCRIPTION
**Pull request by jinquan2019**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

1. Pass user information instance instead of auth token on user change.
2. Rename delegate for auth token for method signature consistency.
3. `UserInformation.accountId == nil` should indicates sign-out.

### CODE REVIEW

**Comment by MatkovIvan** [comment]
Please DO NOT MERGE IT before #1397. It makes conflicts again and we really need to merge that one first.

Thanks.

File: AppCenter/AppCenter/MSUserInformation.h [hunk 1]
```diff
@@ -10,7 +10,11 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * The account identifier for the user.
  */
-@property(nonatomic, copy) NSString *accountId;
+@property(nonatomic, copy, nullable) NSString *accountId;
```
Comment (annakocheshkova) [inline AppCenter/AppCenter/MSUserInformation.h]: is "copy" needed?

File: AppCenter/AppCenter/MSUserInformation.h [hunk 2]
```diff
@@ -10,7 +10,11 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+- (instancetype)initWithAccountId:(nullable NSString *)accountId;
+
+- (BOOL)compareUser:(MSUserInformation *)userInfor;
```
Comment (annakocheshkova) [inline AppCenter/AppCenter/MSUserInformation.h]:
```suggestion
- (BOOL)compareUser:(MSUserInformation *)userInfo;
```

Comment (annakocheshkova) [reply AppCenter/AppCenter/MSUserInformation.h]: Consider more clear naming: compareUserTo

File: AppCenter/AppCenter/MSUserInformation.h (hunk 1 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/MSUserInformation.h]: @annakocheshkova for strings - yes

File: AppCenter/AppCenter/MSUserInformation.m [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "MSUserInformation.h"
+
+@implementation MSUserInformation
+
+- (instancetype)initWithAccountId:(nullable NSString *)accountId {
+  self = [super init];
+  if (self) {
```
Comment (jwallra) [inline AppCenter/AppCenter/MSUserInformation.m]: Use this format and delete line above for consistency

```suggestion
  if ((self = [super init])) {
```

File: AppCenter/AppCenter/MSUserInformation.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSUserInformation : NSObject
```
Comment (jwallra) [inline AppCenter/AppCenter/MSUserInformation.h]: Why are we adding this now when it's just a wrapper for accountId?  We could have waited until we had more user info to store to refactor.  It's probably a good choice to do the refactoring, I just wanted to know the rationale

**Comment by clpolet** [comment]
+1 on @jaelim-ms 's comment. The work here is just about refactoring the delegate methods to provide an `MSUserInformation` instead of an account Id. You would create  an `MSUserInformation` object based on the account Id before calling the delegates.

File: AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.h [hunk 5]
```diff
@@ -18,6 +19,11 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property(nullable, atomic, readonly) NSString *authToken;
 
+/**
+ * The last value of user information.
+ */
+@property(nonatomic, strong, readonly) MSUserInformation *homeUser;
```
[the remark this answers is not in the document]
Comment (jinquan2019) [reply AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.h]: removed

**Comment by jinquan2019** [comment]
> +1 on @jaelim-ms 's comment. The work here is just about refactoring the delegate methods to provide an `MSUserInformation` instead of an account Id. You would create an `MSUserInformation` object based on the account Id before calling the delegates.

Yes, I change the code to apply his comment. Please review again.

**Pull request merged**